### PR TITLE
currentUser: merge actions from state/user

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

This PR is part of the migration of the current user object from `state/users` to `state/current-user` which is a prerequisite to migrate the users store from flux to redux.

* rename `USER_RECEIVE` to `CURRENT_USER_RECEIVE`
* replace `setCurrentUserId` and `receiveUser` with `setCurrentUser`

**Note:** We aren't yet moving the actual current user object to `state/current-user`. This will be part of a subsequent PR.

#### Testing instructions

* the current user object is used throughout calypso and set via boot
* a booting Calypso is a good sign this didn't break anything
* in your browser console compare `getState().currentUser` and `getState().users.items` with `master` which should be the same on this branch

### CODE REVIEW

**Review by jsnajdr — COMMENTED** [review]
This PR should also eliminate the `CURRENT_USER_ID_SET` action type, but it doesn't. It's no longer dispatched anywhere, but the `state.posts.counts` reducer still reacts to it. The current user ID is used when updating the post counts for "Me" and "Everyone" when a post status changes.

**Comment by flootr** [comment]
@jsnajdr would you mind another look (fc0aff2)?

File: client/state/posts/counts/reducer.js [hunk 1]
```diff
@@ -145,8 +145,8 @@ export const counts = ( () => {
 
 				return state;
 			},
-			[ CURRENT_USER_ID_SET ]: ( state, action ) => {
-				currentUserId = action.userId;
+			[ CURRENT_USER_RECEIVE ]: ( state, action ) => {
+				currentUserId = action.user.id;
```
Comment (jsnajdr) [inline client/state/posts/counts/reducer.js]: The `ID` property name is uppercase here: `action.user.ID`.

**Review by jsnajdr — APPROVED** [review]
Thanks for fixing the `ID` also in the unit test :wink: Even posts counts update works :+1: and we're ready to :ship:

**Pull request merged**